### PR TITLE
🔥Refactor Wikipedia

### PR DESCRIPTION
**Pull request by alihaiderrizvi**
Changes made:

- Using google search wrapper to land at Wikipedia website for cross-referencing.
- Refactored get_short_details function.

### CODE REVIEW

File: apis/wikipedia.py [hunk 1]
```diff
@@ -18,92 +27,73 @@ def retrieve_short_description(data: str) -> str:
...
-    DATA = requests.get(url=wikipedia_api_url, params=PARAMS).json()
-    api_data['short description'] = retrieve_short_description(DATA)
+        DATA = requests.get(url=wikipedia_api_url, params=PARAMS).json()
+        api_data['short description'] = retrieve_short_description(DATA)
 
-    return api_data
+        return api_data
+    
+    except Exception as e:
+        return {'message': 'error'}
```
Comment (hxghhhh) [inline apis/wikipedia.py]: Have a more specific error message here 
```suggestion
		  return {'message': '[wikipedia] error getting short description'}
```

File: apis/wikipedia.py [hunk 2]
```diff
@@ -18,92 +27,73 @@ def retrieve_short_description(data: str) -> str:
...
-
-        except Exception as e:
-            break
-
-    return ans
+def find_key(row):
+    key = row.find('th')
+    if key:
+        return key.text
+    return ''
```
Comment (hxghhhh) [inline apis/wikipedia.py]: use `None` here since you are justing doing a basic guard

File: apis/wikipedia.py [hunk 3]
```diff
@@ -18,92 +27,73 @@ def retrieve_short_description(data: str) -> str:
...
+    except Exception as e:
+        return {'message': 'error'}
 
 
-def check_braces(val: str) -> str:
+def find_value(row):
+    value = row.find('td')
+    if value:
+        return value.text
+    return ''
```
Comment (hxghhhh) [inline apis/wikipedia.py]: use `None` here since you are justing doing a basic guard

File: apis/wikipedia.py [hunk 4]
```diff
@@ -5,6 +5,15 @@
 from concurrent.futures import ThreadPoolExecutor
 from constants import wikipedia_api_url
 import os
+from googlesearch import search
+
+
+def search_google(search_param: str) -> str:
+    response = search(search_param, stop=6)
+    
+    for url in response:
```
Comment (hxghhhh) [inline apis/wikipedia.py]: what if there is multiple wikipedia links for a search param? Wouldn't we want to return all of them. In this case we'd only be returning the first one

File: apis/wikipedia.py (hunk 2 above)
Comment (hxghhhh) [reply apis/wikipedia.py]: also write a small test for this

File: apis/wikipedia.py (hunk 3 above)
Comment (hxghhhh) [reply apis/wikipedia.py]: also write a small test for this

**Review by hxghhhh — CHANGES_REQUESTED** [review]
looks proper, but still needs some work

**Review by joelrojo — APPROVED** [review]
Looking good to me. @alihaiderrizvi assuming you're done with this, safe to merge?

**Comment by alihaiderrizvi** [comment]
> Looking good to me. @alihaiderrizvi assuming you're done with this, safe to merge?

Yea I'm done with this! Ready for merge 👍

**Pull request merged**